cmd/util_fp_finder.go

```diff
@@ -28,40 +30,29 @@ func createFpFinderCommand() *cobra.Command {
 		}),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			fpFinder := util.NewFpFinder()
-
-			sortEnabled, err := cmd.Flags().GetBool("sort")
-			if err != nil {
-				logger.Error().Err(err).Msg("Failed to read value for 'sort' flag")
-				return err
-			}
-
-			uniqEnabled, err := cmd.Flags().GetBool("uniq")
-			if err != nil {
-				logger.Error().Err(err).Msg("Failed to read value for 'uniq' flag")
-				return err
-			}
-
 			filenameArg := args[0]
 			if !checkFilePath(filenameArg) {
 				return fmt.Errorf("file %s doesn't exist", filenameArg)
 			}
 
-			return fpFinder.FpFinder(filenameArg, sortEnabled, uniqEnabled)
+			if extendedDictPath != "" && !checkFilePath(extendedDictPath) {
+				return fmt.Errorf("extended dictionary %s doesn't exist", extendedDictPath)
+			}
+
+			return fpFinder.FpFinder(filenameArg, extendedDictPath, englishDictionaryCommitHash)
 		},
 	}
 
 }
 
 func buildFpFinderCommand() {
 	utilCmd.AddCommand(fpFinderCommand)
-	fpFinderCommand.Flags().BoolP("sort", "s", false, "Sort the output alphabetically")
-	fpFinderCommand.Flags().BoolP("uniq", "u", false, "Remove duplicated value from the output")
+	fpFinderCommand.Flags().StringVarP(&extendedDictPath, "extended-dictionary", "e", "", "Absolute or relative path to the extended dictionary")
+	fpFinderCommand.Flags().StringVarP(&englishDictionaryCommitHash, "english-dictionary-commit-hash", "c", "8179fe68775df3f553ef19520db065228e65d1d3", "English dictionary commit hash from Github")
```

Comment:
Load the commit hash from the `toolchain.yaml` configuration in the coreruleset repo (see https://github.com/coreruleset/crs-toolchain/blob/60e0115dae2b2e93fe6342208c6662eec2767c09/configuration/configuration.go#L14). Default to the main branch.

I'd also add the URL to the file on the  main branch to the flag help, so it's easy to find out where to look for the file.

Reply:
I would suggest for now, adding something like into configuration/configuration.go:
```
const defaultDictionaryCommitHash = "the hash"
```
and then returning the default in the `func New` . Then the struct is public so when the flag passes a new hash, just store it.

Reply:
Personally, I like to use an additional boolean to determine the state. For example:
```go
type Dictionary struct {
    CommitHash string
    WasCommitHashSet bool
}
```
That way you can initialize `CommitHash` with some dummy value but don't need to check against it. Instead, you can check the boolean to decide whether to use `master`, or load fro `configuration`.

Reply:
Ping @S0obi